e2e/tests/api-driven/src/permissions/steps.ts

```diff
@@ -1,72 +1,101 @@
 import { After, Before, Given, Then, When, World } from "@cucumber/cucumber";
 import { strict as assert } from "node:assert";
 import { getUser } from "../globalHelpers";
-import { addUserToTeam, cleanup, performGQLQuery, setup } from "./helpers";
-
-interface TestUser {
-  id: number;
-  email: string;
-}
+import {
+  Action,
+  Table,
+  addUserToTeam,
+  cleanup,
+  performGQLQuery,
+  setup,
+} from "./helpers";
 
 export class CustomWorld extends World {
-  user1!: TestUser;
-  user2!: TestUser;
-  teamId1!: number;
-  teamId2!: number;
-  team1Flow!: string;
-  team2Flow!: string;
+  user1Id!: number;
+  user1Email!: string;
+  team1Id!: number;
+  team1FlowId!: string;
+
+  team2Id!: number;
+  user2Id!: number;
+  user2Email!: string;
+  team2FlowId!: string;
+
+  activeUserId!: number;
+  activeUserEmail!: string;
 
   error?: Error = undefined;
-  activeUser!: TestUser;
+  result: any[] | Record<"returning", any[]> | null = null;
```

Comment:
Personally, would love to turn off the type warnings for `unexpected any` to make PRs a bit less noisy - I think we can trust each other to catch actually-unnecessary ones in code reviews ?